<?xml version="1.0"?>
<launch>

    <!-- Specify arguments -->
    <arg name="param_file"                      default="interpreter_params.yaml" />
    <arg name='topic_to_port_motor'             default='/LmotorSpeed' />
    <arg name='topic_to_strbrd_motor'           default='/RmotorSpeed' />
    <arg name='topic_from_heading_controller'   default='heading/control_effort' />
    <arg name='topic_from_speed_controller'     default='speed/control_effort' />

    <!-- Launch PID interpreter -->
    <node pkg="wamv_control" type="pid_interpreter.py" name="interpreter">
        <rosparam command="load" file="$(find wamv_control)/params/$(arg param_file)" />
        <remap from='topic_to_port_motor'            to='$(arg topic_to_port_motor)' />
        <remap from='topic_to_strbrd_motor'          to='$(arg topic_to_strbrd_motor)' />
        <remap from='topic_from_heading_controller'  to='$(arg topic_from_heading_controller)' />
        <remap from='topic_from_speed_controller'    to='$(arg topic_from_speed_controller)' />
   </node>
   
</launch>
